src/Adapter/Meta/UrlSchemaDataConfiguration.php

```diff
@@ -26,35 +26,36 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Meta;
 
-use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
-use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+use PrestaShop\PrestaShop\Adapter\Configuration;
+use PrestaShop\PrestaShop\Adapter\Shop\Context;
+use PrestaShop\PrestaShop\Core\Configuration\AbstractMultistoreConfiguration;
+use PrestaShop\PrestaShop\Core\Feature\FeatureInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
  * Class UrlSchemaDataConfiguration is responsible for validating, updating and retrieving data used in
  * Shop parameters -> Traffix & Seo -> Seo & Urls -> Set Shop URL form field.
  */
-final class UrlSchemaDataConfiguration implements DataConfigurationInterface
+final class UrlSchemaDataConfiguration extends AbstractMultistoreConfiguration
 {
     /**
      * @var array
      */
     private $rules;
 
-    /**
-     * @var ConfigurationInterface
-     */
-    private $configuration;
-
     /**
      * UrlSchemaDataConfiguration constructor.
      *
-     * @param ConfigurationInterface $configuration
+     * @param Configuration $configuration
+     * @param Context $shopContext
+     * @param FeatureInterface $multistoreFeature
      * @param array $rules
      */
-    public function __construct(ConfigurationInterface $configuration, array $rules)
+    public function __construct(Configuration $configuration, Context $shopContext, FeatureInterface $multistoreFeature, array $rules)
```

Comment:
```suggestion
    public function __construct(ConfigurationInterface $configuration, Context $shopContext, FeatureInterface $multistoreFeature, array $rules)
```

Always prefer interface instead of class itself it's better for extension usage